src/Symfony/Component/HttpClient/UriTemplate/CallbackUriTemplateExpander.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\UriTemplate;
+
+/**
+ * This implementation allows you to easily forward the call to a vendor implementation of your choice.
+ */
+class CallbackUriTemplateExpander implements UriTemplateExpanderInterface
+{
+    /**
+     * @param \Closure(string $url, array $variables): string $callback
+     */
+    public function __construct(private readonly \Closure $callback)
+    {
+    }
+
+    public function expand(string $url, array $variables): string
+    {
+        return ($this->callback)($url, $variables);
+    }
+
+    /**
+     * @internal
+     */
+    public static function createFromPopularVendors(): static
+    {
+        if (class_exists(\League\Uri\UriTemplate::class)) {
+            return new self(static fn (string $url, array $variables): string => (new \League\Uri\UriTemplate($url))->expand($variables));
+        }
+
+        if (class_exists(\Rize\UriTemplate::class)) {
+            return new self((new \Rize\UriTemplate())->expand(...));
+        }
+
+        if (class_exists(\GuzzleHttp\UriTemplate\UriTemplate::class)) {
+            return new self(\GuzzleHttp\UriTemplate\UriTemplate::expand(...));
+        }
+
+        throw new \RuntimeException('No URI template vendor found. Run "composer require league/uri", or "composer require rize/uri-template", or "composer require guzzlehttp/uri-template", or use your own URI template expander.');
```

Comment:
let's recommend only one. I'd go for guzzlehttp/uri-template because it has no dependencies and we already have links to this vendor